lib/diagnostic.rb

```diff
@@ -6,37 +6,43 @@ class Diagnostic
   # Write ruby code to return the longest entry in the list supplied.
   # Assume each entry in the list implements a length method that returns
   # a Fixnum.
+
   def longest(list)
-    # Your code goes here
+    list.each{ |prev, e|
+      if e.length > prev.length
+        max = e
+      end
+    }
+    max
   end
```

Comment:
Recall that `.each` is a method that handles individual elements in an Array. Here's the documentation: https://ruby-doc.org/core-2.2.0/Array.html#method-i-each

As such, you have 2 errors here: 1) your input, `list`, is not an array, 2) `.each` only works with 1 thing at a time